### PR TITLE
=doc fix scaladoc warnings #18500

### PR DESCRIPTION
**Pull request by wlk**
Another iteration of fixing Scala/Javadoc warnings, addresses #18500

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

File: akka-http/src/main/scala/akka/http/javadsl/server/directives/PathDirectives.scala [hunk 1]
```diff
@@ -134,7 +134,7 @@ abstract class PathDirectives extends MiscDirectives {
    * }}}
    *
    * For further information, refer to:
-   * [[http://googlewebmastercentral.blogspot.de/2010/04/to-slash-or-not-to-slash.html]]
+   * @see "http://googlewebmastercentral.blogspot.de/2010/04/to-slash-or-not-to-slash.html"
```
Comment (ktoso) [inline akka-http/src/main/scala/akka/http/javadsl/server/directives/PathDirectives.scala]: Are you sure this results in valid scaladoc and javadoc?
I'd think this breaks scaladoc

File: akka-http/src/main/scala/akka/http/javadsl/server/RejectionHandler.scala [hunk 2]
```diff
@@ -134,7 +134,7 @@ abstract class RejectionHandler {
   def handleUnacceptedResponseEncodingRejection(ctx: RequestContext, supported: jl.Iterable[HttpEncoding]): RouteResult = passRejection()
 
   /**
-   * Callback called to handle rejection created by an [[akka.http.scaladsl.server.authentication.HttpAuthenticator]].
+   * Callback called to handle rejection created by an [[akka.http.javadsl.server.values.HttpBasicAuthenticator]].
```
Comment (ktoso) [inline akka-http/src/main/scala/akka/http/javadsl/server/RejectionHandler.scala]: good catch!

File: akka-http/src/main/scala/akka/http/javadsl/server/directives/PathDirectives.scala (hunk 1 above)
Comment (wlk) [inline akka-http/src/main/scala/akka/http/javadsl/server/directives/PathDirectives.scala]: probably should be `@see [["..."]]` checking this now

**Comment by wlk** [comment]
In javadoc generated by "genjavadoc" the links are not clickable anyway, so I have just pasted them directly so at least they won't generate warnings.
In scaladoc, the exptected format is `@see [[..]]` - this makes the link clickable and not generate any warnings

**Comment by ktoso** [comment]
How does the `@see` part show up in scaladoc though? AFAIR it does not parse that, do we have a `"@see"` in the text?

**Comment by wlk** [comment]
this:
```
...
* For further information, refer to:
   * @see [[http://googlewebmastercentral.blogspot.de/2010/04/to-slash-or-not-to-slash.html]]
...
```

shows up as:
see

**Comment by ktoso** [comment]
Hm, weird I'd expect javadoc to understand `@see`, perhaps this should be solved in genjavadoc: https://github.com/typesafehub/genjavadoc/issues/69

**Comment by patriknw** [comment]
LGTM
if you are still looking for how to create external links, I think ordinary `a href` tag works both in scaladoc and javadoc

**Comment by ktoso** [comment]
Failure was unrelated and seems like overall improvement, let's merge this one to keep the queue small.

**Comment by ktoso** [comment]
Thanks @wlk !

**Pull request merged**